space.tf

```diff
@@ -1,5 +1,5 @@
 resource "cloudfoundry_space" "space" {
-  name = "thanos-${random_id.id.hex}"
+  name = var.thanos_space != "" ? var.thanos_space : "thanos-${random_id.id.hex}"
```

Comment:
The space will always be created in this construct, so you will get an error in case the space is created by other parts of your Terraform scripts, or if its an existing space created through CLI. Does that conflict with your use case?

Reply:
Perhaps a `name_postfix` variables which is appended to the existing naming convention creates the desired traceability. Example of this: 

https://github.com/philips-labs/terraform-cloudfoundry-grafana/blob/ee741af50252f7e7de67805f8c394145d824ded5/main.tf#L58

Reply:
Updated the pull-request by changing the implementation to use the common name_postfix convention